### PR TITLE
server: break up the Server.Serve method into some reusable parts

### PR DESCRIPTION
**Pull request by bradfitz**
Updates grpc/grpc-go#75

### CODE REVIEW

**Comment by iamqizhao** [comment]
Sorry, can you resolve the conflicts with #509?

**Comment by bradfitz** [comment]
Done.

File: server.go [hunk 1]
```diff
@@ -264,49 +264,74 @@ func (s *Server) Serve(lis net.Listener) error {
...
+		return
+	}
+	if !s.addConn(st) {
+		c.Close()
+		return
+	}
+	s.serveConnection(st)
+}
+
+func (s *Server) serveConnection(st transport.ServerTransport) {
```
Comment (iamqizhao) [inline server.go]: serveConnection --> serveStreams?

Comment (bradfitz) [inline server.go]: Sure. Will do.

**Comment by bradfitz** [comment]
PTAL

**Comment by iamqizhao** [comment]
LGTM, will merge after travis is done.

**Pull request merged**